src/cli/cluster/configure_base_path_proxy.test.js

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+jest.mock('../../core', () => ({
+  createBasePathProxy: jest.fn(),
+}));
+
+jest.mock('../../server/logging', () => ({
+  setupLogging: jest.fn(),
+}));
+
+import { Server } from 'hapi';
+import { createBasePathProxy as createBasePathProxyMock } from '../../core';
+import { setupLogging as setupLoggingMock } from '../../server/logging';
+import { configureBasePathProxy } from './configure_base_path_proxy';
+
+describe('configureBasePathProxy()', () => {
+  it('returns `BasePathProxy` instance.', async () => {
+    const basePathProxyMock = { configure: jest.fn() };
+    createBasePathProxyMock.mockReturnValue(basePathProxyMock);
+
+    const basePathProxy = await configureBasePathProxy({});
+
+    expect(basePathProxy).toBe(basePathProxyMock);
+  });
+
+  it('correctly configures `BasePathProxy`.', async () => {
+    const configMock = {};
+    const basePathProxyMock = { configure: jest.fn() };
+    createBasePathProxyMock.mockReturnValue(basePathProxyMock);
+
+    await configureBasePathProxy(configMock);
+
+    // Check that logging is configured with the right parameters.
+    expect(setupLoggingMock).toHaveBeenCalledWith(
+      expect.any(Server),
+      configMock
+    );
+
+    const [[server]] = setupLoggingMock.mock.calls;
+    expect(createBasePathProxyMock).toHaveBeenCalledWith({
+      config: configMock,
+      server,
+    });
+
+    expect(basePathProxyMock.configure).toHaveBeenCalledWith({
+      isKibanaPath: expect.any(Function),
+      blockUntil: expect.any(Function),
+    });
+  });
+
+  describe('configured with the correct `isKibanaPath` and `blockUntil` functions.', async () => {
```

Comment:
__note:__ one can say these aren't canonical unit tests as they test too much internals and they will be correct, but I just feel more comfortable with these tests than without them :)